src/UIHandler.h

```diff
@@ -0,0 +1,110 @@
+#pragma once
+#ifndef UIHANDLER_H
+#define UIHANDLER_H
+
+#include "debug.h"
+
+#include "cursesdef.h"
+#include "output.h"
+
+struct IntPair
+{
+    int x;
+    int y;
+};
+
+class UIPanel
+{
+public:
+    virtual std::vector<UIPanel*> getChild() = 0;
+    virtual void addChild(UIPanel *panel) = 0;
+    virtual void removeChild(size_t index) = 0; // Passes back removed child
+
+    virtual IntPair RequestedSize(bool min) = 0;
```

Comment:
The parameter is for getting either a the minimal size or the preferred size. It probably should be an enum.